mage_ai/cache/block_action_object/__init__.py

```diff
@@ -254,13 +254,18 @@ async def initialize_cache_for_all_objects(self) -> None:
                 include_content=True,
             )
 
-        for path_dict in build_repo_path_for_all_projects(mage_projects_only=True).values():
-            repo_path = path_dict['full_path']
+        traversed_paths = dict()
+        for path_dict in build_repo_path_for_all_projects(mage_projects_only=False).values():
+            repo_path = path_dict['root_project_full_path']
```

Comment:
The for loop at Line 258 could possibly be replaced with below (like in this [commit](https://github.com/mage-ai/mage-ai/pull/4347/commits/f017026c2c7f85e3c4d07e5259c3826780ff8d28)).
```
repo_path = base_repo_path()
```